hardware_interface/src/resource_manager.cpp

```diff
@@ -192,6 +204,36 @@ class ResourceStorage
             hardware.get_name().c_str(), interface.c_str());
         }
       }
+
+      if constexpr (std::is_same_v<hardware_interface::Actuator, HardwareT>)
+      {
+        if (hardware_info_map_[hardware.get_name()].is_async)
```

Comment:
maybe a similar style solution could help to have a more maintainable/readable version?

```
    auto load_and_init_actuators = [&](auto & container)
    {
      check_for_duplicates(hardware_info);
      load_hardware<Actuator, ActuatorInterface>(hardware_info, actuator_loader_, container);
      initialize_hardware(hardware_info, container.back());
      import_state_interfaces(container.back());
      import_command_interfaces(container.back());
    };

    if (hardware_info.is_async)
    {
      load_and_init_actuators(async_actuators_);
    }
    else
    {
      load_and_init_actuators(actuators_);
    }
```

maybe not!

Reply:
I switched to std::variant so there aren't any checks anymore, thus it's not needed